src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/misc/ModuleAutoPearl.kt

```diff
@@ -0,0 +1,300 @@
+import net.ccbluex.liquidbounce.config.types.NamedChoice
+import net.ccbluex.liquidbounce.config.types.ToggleableConfigurable
+import net.ccbluex.liquidbounce.event.events.PacketEvent
+import net.ccbluex.liquidbounce.event.handler
+import net.ccbluex.liquidbounce.event.tickHandler
+import net.ccbluex.liquidbounce.features.module.Category
+import net.ccbluex.liquidbounce.features.module.ClientModule
+import net.ccbluex.liquidbounce.features.module.modules.combat.killaura.ModuleKillAura
+import net.ccbluex.liquidbounce.features.module.modules.player.invcleaner.HotbarItemSlot
+import net.ccbluex.liquidbounce.utils.render.trajectory.TrajectoryInfo
+import net.ccbluex.liquidbounce.utils.render.trajectory.TrajectoryInfoRenderer
+import net.ccbluex.liquidbounce.utils.aiming.Rotation
+import net.ccbluex.liquidbounce.utils.aiming.RotationManager
+import net.ccbluex.liquidbounce.utils.aiming.RotationsConfigurable
+import net.ccbluex.liquidbounce.utils.combat.CombatManager
+import net.ccbluex.liquidbounce.utils.combat.shouldBeAttacked
+import net.ccbluex.liquidbounce.utils.inventory.OFFHAND_SLOT
+import net.ccbluex.liquidbounce.utils.inventory.useHotbarSlotOrOffhand
+import net.ccbluex.liquidbounce.utils.item.findHotbarItemSlot
+import net.ccbluex.liquidbounce.utils.kotlin.Priority
+import net.minecraft.entity.Entity
+import net.minecraft.entity.EntityType
+import net.minecraft.entity.projectile.thrown.EnderPearlEntity
+import net.minecraft.item.Items
+import net.minecraft.network.packet.s2c.play.EntitySpawnS2CPacket
+import net.minecraft.util.hit.HitResult
+import net.minecraft.util.math.MathHelper
+import net.minecraft.util.math.Vec3d
+import kotlin.math.*
+
+private const val MAX_SIMULATED_TICKS = 240
+
+/**
+ * I actually don't know how2 turn the player's camera to the direction where i want to throw the ender pearl
+ * without this crutch
+ */
+private const val DIFF_STABILIZATION = 0.2
+
+/**
+ * Auto pearl module
+ *
+ * AutoPearl aims and throws a pearl at an enemies pearl trajectory
+ *
+ * @author sqlerrorthing
+ */
+object ModuleAutoPearl : ClientModule("AutoPearl", Category.MISC, aliases = arrayOf("PearlFollower")) {
+
+    private val mode by enumChoice("Mode", Modes.TRIGGER)
+
+    init {
+        tree(Rotate)
+        tree(Limits)
+    }
+
+    private object Limits : ToggleableConfigurable(this, "Limits", true) {
+        val angle by int("Angle", 180, 0..180, suffix = "°")
+        val activationDistance by float("MinDistance", 8.0f, 0.0f..10.0f, suffix = "m")
+        val destDistance by float("DestinationDistance", 8.0f, 0.0f..30.0f, suffix = "m")
+    }
+
+    private object Rotate : ToggleableConfigurable(this, "Rotate", true) {
+        val rotations = tree(RotationsConfigurable(this))
+    }
+
+    private val combatPauseTime by int("CombatPauseTime", 0, 0..40, "ticks")
+    private val slotResetDelay by intRange("SlotResetDelay", 0..0, 0..40, "ticks")
+
+    private val queue = ArrayQueue<Rotation>()
```

Comment:
What is this queue for?

Reply:
The queue is used to put rotations from PacketEvent to be handled in the Game Tick Handler.